artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/AMQPBrokerConnection.java

```diff
@@ -277,9 +307,11 @@ private void doConnect() {
             for (AMQPBrokerConnectionElement connectionElement : brokerConnectConfiguration.getConnectionElements()) {
                if (connectionElement.getType() == AMQPBrokerConnectionAddressType.MIRROR) {
                   AMQPMirrorBrokerConnectionElement replica = (AMQPMirrorBrokerConnectionElement)connectionElement;
-                  Queue queue = server.locateQueue(replica.getSourceMirrorAddress());
 
-                  connectSender(queue, ProtonProtocolManager.MIRROR_ADDRESS, (r) -> AMQPMirrorControllerSource.validateProtocolData(r, replica.getSourceMirrorAddress()));
+                  Queue queue = server.locateQueue(getMirrorSNF(replica));
+
+                  connectSender(queue, queue.getName().toString(), mirrorControllerSource::setLink, (r) -> AMQPMirrorControllerSource.validateProtocolData(protonProtocolManager.getReferenceIDSupplier(), r, getMirrorSNF(replica)), server.getNodeID().toString(),
+                                new Symbol[]{AMQPMirrorControllerSource.MIRROR_CAPABILITY}, new Symbol[]{AMQPMirrorControllerSource.MIRROR_CAPABILITY});
```

Comment:
```suggestion
                                new Symbol[]{AMQPMirrorControllerSource.MIRROR_CAPABILITY}, null});
```

I dont think the target capability is being used or needed anymore?

Reply:
you're right. thx